### ISSUE TITLE
a old bug ;o

### ISSUE THREAD

**Issue by AconTM** [tastybento/ASkyBlock-Bugs-N-Features#550]
Hi,

a long time ago my system has a bug with Island-Fly Permission.. I Fixed it but now the bug
is back ;o i dont know what i have to do :/ 

"Bug": 

When a player comes from another Map, he cant Fly on his Island (with askyblock.islandfly).. Can you pls tell me how to fix this ^^ 

Plugins:

PermissionsEx
Worldguard
Multiverse

**Comment by tastybento** [comment]
What version of the plugin and server are you using?

**Comment by AconTM** [comment]
lates version & Spigot 1.11.2

**Comment by AconTM** [comment]
Login on Island = Player can fly
Player go to spawn = Player can't fly (obviously)
Player go back to his Island = Player can't fly again (this is the Problem) 

Sidenote= My spawn is on another world

**Comment by tastybento** [comment]
I can't replicate this right now. I tried with the following plugins on 1.11.2:
PermissionsEx v1.23.4, WorldEdit v6.1.8-SNAPSHOT;cd4729f, Vault, ASkyBlock, Essentials v2.0.1-b488, Multiverse-Core v2.6.0-SNAPSHOT-b736

Sequence tried:

1. Login, player can fly
2. Do /mvtp world to teleport to the other world (how do you do it?)
3. Do /is to teleport back to island - can still fly

Player is not Op and has these perms:

1) multiverse.* (own)
2) askyblock.islandfly (own)

The first one is just so I can teleport out of the world.

If you try the above, what do you see? What perms does your player have? How do you tp to spawn and back?

**Comment by AconTM** [comment]
btw, i have found another Bug:

When i login on my Island & i have the perm to Islandfly --> On my Server users where Teleport to the Spawn on login & if i was on my Island i can fly on the Spawn ._.

I hope you understand my broken English ^^

**Comment by AconTM** [comment]
#Update 

I fixed the "Fly-Bug" on spawnTP with a simple waiting time for the Teleport (2 ticks after join)

**Comment by tastybento** [comment]
How does a player go to spawn from the ASKyBlock world on your server? What command does the player use?

**Comment by AconTM** [comment]
EssentialsSpawn Plugin

**Comment by AconTM** [comment]
Hey ^^ i have a awnser how my Bug "work":

If Players Change the World and in this other world is GameMode "ADVENTURE" as Default, they cant fly if they go back to the Island

**Comment by tastybento** [comment]
I'll try and see if I can replicate again.

**Comment by AconTM** [comment]
how looks?

**Comment by tastybento** [comment]
I can't replicate sorry. 

I went to a world and did game mode 2 (Adventure) and then /is. I could still fly.

Does this match you?

**Comment by AconTM** [comment]
No.. Have you Multiverse Permissions? it only works if i am in gamemode 0..

**Comment by AconTM** [comment]
The Problem is, with the Permission "mv.bypass.gamemode.<WORLD>" are you skipping the change of the "allow_fly" from true to false :/ So with permissions, it works, but without, it doesnt work :/
